src/util.cc

```diff
@@ -2058,6 +2058,13 @@ int time_compare(struct timeval* tv_a, struct timeval* tv_b)
 		return tv_a->tv_sec - tv_b->tv_sec;
 	}
 
+double curr_CPU_time()
+	{
+	struct timespec ts;
+	clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &ts);
```

Comment:
For the purpose of script profiling I wonder this should be using `CLOCK_THREAD_CPUTIME_ID` to get the cpu time solely for the main thread rather than the the whole process.

Unclear how much it matters, but there could be busy caf and logger threads running in parallel next to script execution that may skew the resulting times around.

Reply:
I can see that going either way.  Over time, those busy additional threads might correlate with the script's own activity, in which case it's a plus.  OTOH, if they don't, and are bursty, then yeah it's a minus.

How portable is `CLOCK_THREAD_CPUTIME_ID` ?  Assuming it's readily available, another approach would be to provide both CPU times ... though that starts making the profile a bit clunky.

Reply:
> Over time, those busy additional threads might correlate with the script's own activity, in which case it's a plus. OTOH, if they don't, and are bursty, then yeah it's a minus.

Scenario I had in mind would be some function doing a `Log::write()`. The actual JSON rendering/gzip/.. happening in a parallel running  logger thread while the main thread moved on executing a different/unrelated handler/function.

But given PROCESS_CPUTIME_ID provides useful results, maybe just something to keep in mind at this point as it might not matter in practice.

> How portable is CLOCK_THREAD_CPUTIME_ID ? Assuming it's readily available, another approach would be to provide both CPU times ... though that starts making the profile a bit clunky.

Mostly not sure about OSX, probably okay on Linux/BSD I guess. Sticking with PROCESS_CPUTIME_ID is probably better than clunky.